modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -170,10 +204,12 @@ private static void initializeSlotMapForQueue(String queueName)
         int numberOfMessages = messageList.size();
 
         //setting up timer to print restoring message count and database read count
-        numberOfReadsFromDatabase++;
-        restoreSlotMappingCounter = restoreSlotMappingCounter + messageList.size();
+        databaseReadsCounter++;
+        restoreMessagesCounter = restoreMessagesCounter + messageList.size();
+        databaseReadsCounterMap.put(queueName, databaseReadsCounter);
+        restoreMessagesCounterMap.put(queueName, restoreMessagesCounter);
         Timer timer = new Timer();
-        scheduleTimerToPrintCounter(timer);
+        scheduleTimerToPrintCounter(timer, queueName);
```

Comment:
I think it would be great if we can use a `ScheduledExecutorService` instead of a `Timer`. It's the modern API for this type of tasks.